compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/types/BLangMemberTypeNode.java

```diff
@@ -0,0 +1,91 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.wso2.ballerinalang.compiler.tree.types;
+
+import org.ballerinalang.model.elements.Flag;
+import org.ballerinalang.model.tree.AnnotatableNode;
+import org.ballerinalang.model.tree.AnnotationAttachmentNode;
+import org.ballerinalang.model.tree.NodeKind;
+import org.wso2.ballerinalang.compiler.tree.BLangAnnotationAttachment;
+import org.wso2.ballerinalang.compiler.tree.BLangNodeAnalyzer;
+import org.wso2.ballerinalang.compiler.tree.BLangNodeTransformer;
+import org.wso2.ballerinalang.compiler.tree.BLangNodeVisitor;
+
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+
+
+/**
+ * {@code BLangUnionTypeNode} represents a non-rest member of a tuple.
+ *
+ * @since 2201.2.0
+ */
+public class BLangMemberTypeNode extends BLangType implements AnnotatableNode {
+
+    public BLangType typeNode;
+    public List<BLangAnnotationAttachment> annAttachments;
+    public Set<Flag> flagSet;
+
+    public BLangMemberTypeNode() {
+        this.annAttachments = new ArrayList<>();
```

Comment:
Shall we keep this `annAttachments` null and not initialise to an empty ArrayList in ctor? I believe even in the current implementation, this gets replaced with a new list. I believe without initializing the list encourage other programmers to initialize the list with a initial capacity because now the programmers will initialize it in a place where they have more information.